maths/three_sum.py

```diff
@@ -0,0 +1,45 @@
+def three_sum(nums: list[int]) -> list[list[int]]:
+    """
+    Find all unique triplets in a sorted array of integers that sum up to zero.
+
+    Args:
+        nums (list[int]): A sorted list of integers.
+
+    Returns:
+        list[list[int]]: A list of lists containing unique triplets that sum up to zero.
```

Comment:
[Mypy](mypy.readthedocs.io) can test data types in the function signature, but not in the comments.

Do not repeat the datatypes in both places because readers will be confused if one is changed and the other is not.

```suggestion
        nums: A sorted list of integers.

    Returns:
       A list of lists containing unique triplets that sum up to zero.
```

Reply:
when I removed pre-commit.ci - pr is failing